### ISSUE TITLE
Avoid auxiliary derived table in SQLite GENERATE_SERIES emulation

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#11582]
SQLite doesn't support its `GENERATE_SERIES` emulation:

```sql
select generate_series 
from (
  with recursive generate_series(v) as (
    select * 
    from (select ?) x 
    union all 
    select * 
    from (
      select (v + 1) 
      from generate_series 
      where v < ?
    ) x
  ) 
  select v as generate_series 
  from generate_series
) as generate_series 
order by 1
```

There's an error:

> [SQLITE_ERROR] SQL error or missing database (recursive reference in a subquery: generate_series)

We should push recursive CTE to the top level or find another solution for `GENERATE_SERIES`

**Comment by lukaseder** [comment]
Hmm, no, this seems to be merely caused by the auxiliary derived table `x`. This works just fine:

```sql
select generate_series 
from (
  with recursive generate_series(v) as (
    select * 
    from (select 1) x 
    union all 
    select (v + 1) 
    from generate_series 
    where v < 10
  ) 
  select v as generate_series 
  from generate_series
) as generate_series 
order by 1
```

**Comment by lukaseder** [comment]
Nested set operations getting in the way again (any of #3579, #6431, #7222)

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.15.0 and 3.14.9 (#11600)
